supernode/daemon/mgr/scheduler/manager.go

```diff
@@ -193,7 +193,7 @@ func (sm *Manager) tryGetPID(ctx context.Context, taskID string, pieceNum int, p
 		}
 
 		// if the v is in the blackList, try the next one.
-		blackInfo, err := sm.progressMgr.GetBlackInfoByPeerID(ctx, peerIDs[i])
+		blackInfo, _ := sm.progressMgr.GetBlackInfoByPeerID(ctx, peerIDs[i])
```

Comment:
Add the following code instead of ignoring err.

```
if err != nil && !errorType.IsDataNotFound(err) {
	logrus.Errorf("failed to get blackInfo for peerID %s: %v", peerIDs[i], err)
	continue
}
```